payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -288,8 +288,10 @@ internal class PaymentSheetViewModel @Inject internal constructor(
     }
 
     fun checkout(checkoutIdentifier: CheckoutIdentifier) {
-        // Clear out any previous errors before setting the new button to get updates.
-        _viewState.value = PaymentSheetViewState.Reset(null)
+        if (this.checkoutIdentifier != checkoutIdentifier) {
+            // Clear out any previous errors before setting the new button to get updates.
+            _viewState.value = PaymentSheetViewState.Reset(null)
```

Comment:
now that you already touched it: `PaymentSheetViewState.Reset()`

Reply:
Done